src/io_posix.c

```diff
@@ -167,49 +167,32 @@ udx__on_writable (udx_socket_t *socket) {
     int nsent = rc > 0 ? rc : 0;
 
     if (rc < 0 && rc != UV_EAGAIN) {
-      nsent = pkts; // something errored badly, assume all packets sent and lost
+      nsent = npkts; // something errored badly, assume all packets sent and lost
     }
 
-    int unsent = pkts - nsent;
+    int unsent = npkts - nsent;
 
-    while (unsent > 0) {
-      // restore an unsent packet
-      udx__fifo_undo(fifo);
-      if (udx__fifo_peek(fifo) != NULL) {
-        unsent--;
-      }
+    // cancel packets in reverse !
+    for (int i = npkts; i > npkts - unsent; i--) {
+      __builtin_trap();
```

Comment:
what does this do?

Reply:
__builtin_trap() is leftover debug code, fixing.